src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts

```diff
@@ -0,0 +1,195 @@
+  ) {
+    super(platform, navController, authenticationProvider);
+  }
+
+  ngOnInit(): void {
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+    const category$ = currentTest$.pipe(
+      select(getTestCategory),
+    );
+    // TODO - PREP-AMOD2 - replace test category
+    this.pageState = {
+      seriousFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data =>
+          this.faultSummaryProvider.getSeriousFaultsList(data, TestCategory.BE)
+          .map(fault => fault.competencyIdentifier)),
+      ),
+      // TODO - PREP-AMOD2 - replace test category
+      dangerousFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data =>
+          this.faultSummaryProvider.getDangerousFaultsList(data, TestCategory.BE)
+          .map(fault => fault.competencyIdentifier)),
+      ),
+      // TODO - PREP-AMOD2 - replace test category
+      drivingFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE)),
+      ),
+      drivingFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        withLatestFrom(category$),
+        map(([testData, category]) => {
+          return this.faultCountProvider.getDrivingFaultSumCount(category as TestCategory, testData);
+        }),
+      ),
+      etaFaults$: currentTest$.pipe(
+        select(getTestData),
+        select(getETA),
+      ),
+      ecoFaults$: currentTest$.pipe(
+        select(getTestData),
+        select(getEco),
+      ),
+      testResult$: currentTest$.pipe(
+        select(getTestOutcome),
+      ),
+      conductedLanguage$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getConductedLanguage),
+      ),
+      candidateName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getUntitledCandidateName),
+      ),
+    };
+
+    const { testResult$, etaFaults$, ecoFaults$, conductedLanguage$ } = this.pageState;
+
+    this.subscription = merge(
+      testResult$.pipe(map(result => this.outcome = result)),
+      etaFaults$.pipe(
+        map((eta) => {
+          this.hasPhysicalEta = eta.physical;
+          this.hasVerbalEta = eta.verbal;
+        }),
+      ),
+      ecoFaults$.pipe(
+        map((eco) => {
+          this.adviceGivenControl = eco.adviceGivenControl;
+          this.adviceGivenPlanning = eco.adviceGivenPlanning;
+        }),
+      ),
+      conductedLanguage$.pipe(tap(value => configureI18N(value as Language, this.translate))),
+    ).subscribe();
+  }
+
+  ionViewDidEnter(): void {
+    this.store$.dispatch(new DebriefViewDidEnter());
+  }
+
+  ionViewDidLeave(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+
+    }
+  }
+
+  endDebrief(): void {
+    this.store$.dispatch(new EndDebrief());
+    if (this.outcome === 'Pass') {
```

Comment:
magic string - is there a definition anywhere of pass and fail?

Reply:
I don't think so I can't seem to find one if there is